File: datasets/rs-keep-code_davinci_001_temp_0.2/HumanEval_15_string_sequence.rs
/// Return a string containing space-delimited numbers starting from 0 upto n inclusive.
/// >>> string_sequence(0)
/// '0'
/// >>> string_sequence(5)
/// '0 1 2 3 4 5'
fn string_sequence(n: isize) -> String {
    let mut result = String::new();
    for i in 0..n+1 {
        result.push_str(&i.to_string());
        result.push(' ');
    }
    result.pop();
    result

}

fn main() {
    let candidate = string_sequence;
    assert_eq!(candidate(0), String::from("0"));
    assert_eq!(candidate(3), String::from("0 1 2 3"));
    assert_eq!(candidate(10), String::from("0 1 2 3 4 5 6 7 8 9 10"));
}

File: datasets/lua-keep-code_davinci_001_temp_0.2/HumanEval_15_string_sequence.lua
-- Return a string containing space-delimited numbers starting from 0 upto n inclusive.
-- >>> string_sequence(0)
-- '0'
-- >>> string_sequence(5)
-- '0 1 2 3 4 5'
local function string_sequence(n)
  local s = ''
  for i = 0, n do
    s = s .. i .. ' '
  end
  return s
end

lu = require('luaunit')

function test_humaneval()
local candidate = string_sequence
    lu.assertEquals(candidate(0), '0')
    lu.assertEquals(candidate(3), '0 1 2 3')
    lu.assertEquals(candidate(10), '0 1 2 3 4 5 6 7 8 9 10')
end

os.exit(lu.LuaUnit.run())